tests/test_core/test_storage.py

```diff
@@ -68,6 +68,36 @@ def test_get_custom_folder(local_storage_manager: StorageManager, project_folder
     assert local_storage_manager.get_folder("eopatches", full_path=True) == abs_path
 
 
+@pytest.mark.parametrize("config_profile", [None, "", "nonexistent-config-profile"])
+@pytest.mark.parametrize("env_profile", [None, "", "nonexistent-env-profile"])
+def test_aws_profile(aws_storage_config: RawConfig, config_profile: Optional[str], env_profile: Optional[str]):
+    """Checks different combinations of profile being set with a config parameter and environmental variable. Checks
+    also that config parameter takes priority over environmental variable."""
+
+    for parameter_key, parameter_value, config_dict in [
+        ("aws_profile", config_profile, aws_storage_config),
+        ("AWS_PROFILE", env_profile, os.environ),
+    ]:
+        if parameter_value is not None:
+            config_dict[parameter_key] = parameter_value
+        elif parameter_key in config_dict:
+            del config_dict[parameter_key]
+
+    try:
+        expected_profile = config_profile if config_profile is not None else env_profile
+        if expected_profile:
+            with pytest.raises(ProfileNotFound) as exception_info:
+                StorageManager.from_raw_config(aws_storage_config)
+
+            assert str(exception_info.value) == f"The config profile ({expected_profile}) could not be found"
+        else:
+            storage = StorageManager.from_raw_config(aws_storage_config)
+            assert storage.config.aws_profile == expected_profile
+    finally:
+        if "AWS_PROFILE" in os.environ:
+            del os.environ["AWS_PROFILE"]
```

Comment:
wouldn't this mess up the users profile settings?

Reply:
No. At the start of Python process, env variables from external environment are copied to Python process. But when Python process ends, any env variable set or changed by Python process are not applied back to external environment. Hence nothings outside the scope of tests will get messed up. But we have to do this step, otherwise this would mess up other tests.

Reply:
ah, as long as it doesnt mess up the environment outside of python this is fine